static/app/views/performance/transactionSummary/transactionTags/tagsHeatMap.tsx

```diff
@@ -166,13 +252,178 @@ const TagsHeatMap = (
+          {({isOpen, getMenuProps, actions}) => {
+            const onChartClick = bucket => {
+              const htmlEvent = bucket.event.event;
+              // Make a copy of the dims because echarts can remove elements after this click happens.
+              // TODO(k-fish): Look at improving this to respond properly to resize events.
+              const virtualRef = new VirtualReference(htmlEvent.target);
+              setChartElement(virtualRef);
+
+              const newTransactionEventView = eventView.clone();
+              const [_, tagValue] = bucket.value;
+
+              if (histogramBucketInfo) {
+                const row = tableData.data[bucket.dataIndex];
+                const currentBucketStart = parseInt(
+                  `${row[histogramBucketInfo.histogramField]}`,
+                  10
+                );
+                const currentBucketEnd =
+                  currentBucketStart + histogramBucketInfo.bucketSize;
+
+                newTransactionEventView.additionalConditions.setTagValues(
+                  'transaction.duration',
+                  [`>=${currentBucketStart}`, `<${currentBucketEnd}`]
+                );
+              }
+
+              newTransactionEventView.additionalConditions.setTagValues(tagKey, [
+                tagValue,
+              ]);
 
-        <HeatMapChart visualMaps={visualMaps} series={series} {...chartOptions} />
+              setTransactionEventView(newTransactionEventView);
+
+              if (!isMenuOpen) {
+                actions.open();
+              }
+            };
+
+            return (
+              <React.Fragment>
+                {ReactDOM.createPortal(
+                  <div>
+                    {chartElement ? (
+                      <Popper referenceElement={chartElement} placement="bottom">
+                        {({ref, style, placement}) => (
+                          <StyledDropdownContainer
+                            ref={ref}
+                            style={style}
+                            className="anchor-middle"
+                            data-placement={placement}
+                          >
+                            <StyledDropdownContent
+                              {...getMenuProps({
+                                className: classNames('dropdown-menu'),
+                              })}
+                              isOpen={isOpen}
+                              alignMenu="right"
+                              blendCorner={false}
+                            >
+                              <TagTransactionsQuery
+                                query={transactionEventView.getQueryWithAdditionalConditions()}
+                                location={location}
+                                eventView={transactionEventView}
+                                orgSlug={organization.slug}
+                                limit={3}
+                                referrer="api.performance.tag-page.heat-map"
```

Comment:
Since this is hitting the `eventsv2` endpoint, you'll want to register this referrer [here](https://github.com/getsentry/sentry/blob/ef35f39d208662f5d32e9a99a603d58132233357/src/sentry/api/endpoints/organization_events.py#L14) otherwise it'll be overwritten with the generic referrer.

Reply:
I didn't know that, thanks